runtime/doc/lua.txt

```diff
@@ -34,20 +34,32 @@ Nvim ever ships with Lua 5.4+, a Lua 5.1 compatibility shim will be provided
 so that old plugins continue to work transparently.
 
                                                                   *lua-luajit*
-Nvim is built with luajit on platforms which support it, which provides
-extra functionality. Lua code in |init.lua| and plugins can assume its presence
-on installations on common platforms. For maximum compatibility with less
-common platforms, availability can be checked using the `jit` global variable: >lua
+On supported platforms, Nvim is built with LuaJIT, which provides extra
+functionality (compared to PUC Lua) such as "bit" and various utilities in
+`$VIMRUNTIME/lua/jit/`. Lua code in |init.lua| and plugins can assume its
+presence on many platforms, but for maximum compatibility should check the
+`jit` global variable: >lua
   if jit then
     -- code for luajit
   else
     -- code for plain lua 5.1
   end
 <
                                                                   *lua-bit*
-In particular, the luajit "bit" extension module is _always_ available.
-A fallback implementation is included when nvim is built with PUC Lua 5.1,
-and will be transparently used when `require("bit")` is invoked.
+The LuaJIT "bit" extension module is _always_ available: when built with PUC
+Lua, Nvim includes a fallback implementation which provides `require("bit")`.
+
+                                                                  *lua-profile*
+To profile Lua code (with LuaJIT-enabled Nvim), the basic steps are: >lua
+    -- Start a profiling session:
+    require('jit.p').start('i1', '/tmp/profile')
+
+    -- Perform arbitrary tasks (use plugins, scripts, etc.) ...
+
+    -- Stop the session. Profile is written to /tmp/profile.
+    require('jit.p').stop()
+
+See `$VIMRUNTIME/lua/jit/p.lua` for details.
```

Comment:
Heh? This location seems incorrect.

Reply:
Yes. This is a LuaJIT(-only) module.

Reply:
https://luajit.org/ext_profiler.html is the best link there is.

Reply:
I would personally remove this line, there's no consistent location where you'd find this file. It depends on `package.path`, which itself depends entirely on how luajit was built and installed. A link to the website is IMO acceptable.

Reply:
If we get reports that users can't find this we can revisit. There is no reason to remove a useful hint for most users.

Reply:
Do I count as a user lol? I mean it, `$VIMRUNTIME/lua/jit/p.lua` is more likely to not exist than it is to exist, I'm fairly sure it's not a useful hint.

Reply:
updated